src/main/scala/Main.scala

```diff
@@ -77,8 +82,10 @@ class SharedAxisCharts extends VBox {
     }
 
     charts.foreach(chart => {
-      chart.chart.xLower.value = newLower
-      chart.chart.xUpper.value = newUpper
+      chart.chart.xLower.value = newXLower
+      chart.chart.xUpper.value = newXUpper
+      chart.chart.yLower.value = newYLower
+      chart.chart.yUpper.value = newYUpper
```

Comment:
Consider only re-assigning the values when they change - can you put these in the relevant if-blocks?